tuned/verifylog.py

```diff
@@ -0,0 +1,103 @@
+
+import json
+
+class VerifyLog:
+
+	obj = None
+
+	def __init__(self, log={}):
+		self.vLog = log
+		return
+
+	@classmethod
+	def get_obj(self, log={}):
+		"""
+		Create and return a singleton object.
+		"""
+		if (not self.obj):
+			self.obj = VerifyLog(log)
+		return self.obj
+
+	def add_log(self, iname, kname, current="",
+                    expected="", devices=[], result=""):
+
+		if (iname not in self.vLog.keys()):
+			self.vLog[iname] = []
+
+		self.vLog[iname].append({kname: {"current": current,
+                                         "expected": expected,
+                                         "devices": devices,
+                                         "result": result}})
+		return
+
+	def get_log(self):
+		return self.vLog
+
+	def put_log(self):
+		vlog = self.vLog
+		self.vLog = {}
+		return vlog
+
+	def _display_device(self, log):
+		ret = True
+		for l in log:
+			for opt, val in l.items():
+				print(" %s=%s" % (opt, val["expected"]))
+				for dev in val["devices"]:
+					for k, v in dev.items():
+						cv = v["current"]
+						rs = v["result"]
+						if (not rs):
+							print("  %s=%s" % (k, cv))
+						if (ret):
+							ret = rs
+		return ret
+
+	def _display_non_device(self, log):
+		ret = True
+		for l in log:
+			for opt, val in l.items():
+				cv = val["current"]
+				ev = val["expected"]
+				rs = val["result"]
+				print(" %s=%s"
+					% (opt, cv), end="\n" if rs else ", ")
+				if (not rs):
+					print("[Expected: %s]" % ev)
+					ret = rs
+		return ret
+
+	def _display_modules(self, log):
+		ret = True
+		for l in log:
+			mod, val = list(l.items())[0]
+			print(" %s: %s" % (mod, val["current"]))
+			ret = val["result"] if (ret) else False
+
+		return ret
+
+	def _display_net(self, log):
+		ret = True
+		for l in log:
+			for opt, val in l.items():
+				if (val["current"]):
+					rs = self._display_non_device([l])
+				elif (not val["current"]):
+					rs = self._display_device([l])
+				if (ret):
+					ret = rs
+		return ret
+
+	def display(self):
+		ret = True
+		instances = json.loads(self.put_log())
+		for key in instances.keys():
+			print("[%s]" % key)
+			if key in ["modules", "script"]:
```

Comment:
I still have problem with the harcoded plugin names. If the log output is different from the "net", could the plugin itself set the logging type? My point is when somebody adds new plugin (or 3rd party upstream unsupported plugin) she or he shouldn't edit verifylog.py (in case of 3rd party plugins she or he even cannot).

Reply:
* Removed the _display_modules() routine and related check if key in ["modules", "script"].
Now there's no check for hard coded plugin names.

* If somebody adds a new plugin, calling vlog.add_log() should help it to display verification logs.

Thank you.